### PR TITLE
fix: implement YamlModel specific errors

### PR DESCRIPTION
**Pull request by jsolaas**
YamlError gives additional info about the error in yaml, in addition to ensuring a consistent interface across different yaml readers.

### CODE REVIEW

File: src/libecalc/presentation/yaml/file_context.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    start: FileMark
+    end: Optional[FileMark] = None
+    name: Optional[str] = None
+
+    def __str__(self) -> str:
+        file_context_string = ""
+        if self.name is not None:
+            file_context_string += f" in '{self.name}',"
+
+        file_context_string += str(self.start)
```
Comment (TeeeJay) [inline src/libecalc/presentation/yaml/file_context.py:27]: not include end, if set?

File: src/tests/libecalc/presentation/yaml/yaml_models/test_yaml_model_errors.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+from libecalc.presentation.yaml.yaml_entities import ResourceStream
+from libecalc.presentation.yaml.yaml_models.exceptions import YamlError
+from libecalc.presentation.yaml.yaml_models.yaml_model import YamlModel, YamlModelType
+
+invalid_models = [
+    (
+        """
+`INSTALLATIONS: []
+""",
+        "invalid_token_start",
```
Comment (TeeeJay) [inline src/tests/libecalc/presentation/yaml/yaml_models/test_yaml_model_errors.py:13]: should this message be presented to users? Should be consider making it more understandable? (if so)

File: src/libecalc/presentation/yaml/file_context.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+class FileMark:
+    line_number: int
+    column_number: int
+
+    def __str__(self) -> str:
+        return f" line {self.line_number}, column {self.column_number}"
+
+
+@dataclass
+class FileContext:
```
Comment (TeeeJay) [inline src/libecalc/presentation/yaml/file_context.py:17]: can this also be used for other resources than yaml, such as csv? Relevant?

File: src/tests/libecalc/presentation/yaml/yaml_models/test_yaml_model_errors.py (hunk 2 above)
Comment (jsolaas) [reply src/tests/libecalc/presentation/yaml/yaml_models/test_yaml_model_errors.py:13]: That's not the expected value in the test, it is just a description/name of the model. This test only makes sure the error raised is YamlError. I haven't done much with the error messages, so we will show what pyyaml/ruamel reports. That may not be understandable to users.

File: src/libecalc/presentation/yaml/file_context.py (hunk 1 above)
Comment (jsolaas) [reply src/libecalc/presentation/yaml/file_context.py:27]: Not sure, seems a bit confusing to me. We might have to be more specific in the errors here, as sometimes start may mean "the error is here" and other times it means the error is between start and end

File: src/libecalc/presentation/yaml/file_context.py (hunk 3 above)
Comment (jsolaas) [reply src/libecalc/presentation/yaml/file_context.py:17]: Sure, that would make sense to me. I'm hoping we can improve on this when looking into validation (of csv resources)

**Pull request merged**